### PR TITLE
thrax: drop GCC dependency

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/thrax.rb [hunk 1]
```diff
@@ -28,19 +29,15 @@ class Thrax < Formula
   depends_on "libtool" => :build
 
   depends_on "openfst"
-
-  on_linux do
-    depends_on "gcc"
-    depends_on "python@3.10"
-  end
+  depends_on "python@3.10"
```
Comment (carlocab) [inline Formula/thrax.rb:32]: Why not
```ruby
uses_from_macos "python"
```
?

Comment (danielnachun) [reply Formula/thrax.rb:32]: I'll change to that.  There's no reason I am aware of that it has to be a specific version of Python 3.

File: Formula/thrax.rb [hunk 2]
```diff
@@ -28,19 +29,15 @@ class Thrax < Formula
   depends_on "libtool" => :build
 
   depends_on "openfst"
-
-  on_linux do
-    depends_on "gcc"
-    depends_on "python@3.10"
-  end
+  uses_from_macos "python"
```
Comment (carlocab) [inline Formula/thrax.rb:32]:
```suggestion
  uses_from_macos "python", since: :catalina
```

File: Formula/thrax.rb [hunk 3]
```diff
@@ -28,19 +29,15 @@ class Thrax < Formula
...
 
   fails_with gcc: "5"
 
   def install
     system "autoreconf", "--force", "--install", "--verbose"
     system "./configure", *std_configure_args
     system "make", "install"
-    rewrite_shebang detected_python_shebang, bin/"thraxmakedep" if OS.linux?
+    rewrite_shebang detected_python_shebang, bin/"thraxmakedep"
```
Comment (carlocab) [inline Formula/thrax.rb:40]:
```suggestion
    rewrite_shebang detected_python_shebang(use_python_from_path: true), bin/"thraxmakedep"
```

**Review by iMichka — APPROVED** [review]

**Pull request closed without merging**